### ISSUE TITLE
Why not having all external sources in a single repo

### ISSUE THREAD

**Issue by Zeioth** [gbprod/none-ls-luacheck.nvim#3]
@gbprod suggestion:

* It's ok if not all external sources live in none-ls-extras.nvim.
* But I would suggest having all the external sources authored by you in a single repo.

## Practical example
Let's say currently, I want to load all your external sources using [none-ls-autoload.nvim](https://github.com/Zeioth/none-ls-autoload.nvim).

```lua
{
  "zeioth/none-ls-autoload.nvim",
  event = "BufEnter",
  dependencies = {
    "williamboman/mason.nvim",
    "nvimtools/none-ls-extras.nvim", -- To install external sources from a repo.
    "gbprod/none-ls-shellcheck.nvim",
    "gbprod/none-ls-luacheck.nvim",
    "gbprod/none-ls-php.nvim",
    "gbprod/none-ls-psalm.nvim",
    "gbprod/none-ls-ecs.nvim"
  },
  opts = {
    external_sources = { -- To indicate where to find a external source.
      -- diagnostics
      'none-ls-shellcheck.diagnostics',
      'none-ls-luacheck.diagnostics.luacheck',
      'none-ls-php.diagnostics.php',
      'none-ls-psalm.diagnostics',

      -- formatting
      'none-ls-ecs.formatting',
    },
  },
},
```

We can observe:

* Several dependencies had to be installed (which is verbose).
* The external source path is different on each of your sources, which can lead to confusion.
* source paths don't currently follow the none-ls path convention of `package_name.method.mason_package_name`.

## Respecting path naming conventions
Instead, the directory structure should be conventional like this

```lua
'gbprob-none-ls.diagnostics.shellcheck',
'gbprob-none-ls.diagnostics.luacheck',
'gbprob-none-ls.diagnostics.php,
'gbprob-none-ls.diagnostics.psalm',

-- formatting
'gbprob-external-sources.formatting.ecs',
```

Otherwise `none-ls-autoload` (or any other plugin) won't be able to load it.

**Comment by Zeioth** [comment]
I've done it already: https://github.com/Zeioth/none-ls-external-sources.nvim

**Comment by gbprod** [comment]
Hi @Zeioth !

I think I was the first to create repositories for external sources, so I didn't have any convention and I did that to continue using those sources after their removal from main repo.

If you have created a repo with all my sources and you want to maintain this, maybe I should deprecate my repositories and point to yours.
What do you think about that ?

**Comment by Zeioth** [comment]
@gbprod you decide. This was a requirement to implement https://github.com/Zeioth/none-ls-autoload.nvim

if you want we can both maintain https://github.com/Zeioth/none-ls-external-sources.nvim

and it's probably going to be less of a headache for users.

**Comment by gbprod** [comment]
I agree, it's going to be less of a headache for users. I will deprecate my repos in favor of yours (just let me some time ;) )
Just a question, why not adding these sources to https://github.com/nvimtools/none-ls-extras.nvim ?

Thanks for your work

**Comment by Zeioth** [comment]
That would work too @gbprod, I just created the repo so we can start using https://github.com/Zeioth/none-ls-autoload.nvim

https://github.com/Zeioth/none-ls-external-sources.nvim

I already include your sources to my repo, so feel free to pr them to none-extra if you want too
